connaisseur/config.py

```diff
@@ -40,6 +40,11 @@ def __init__(self):
 
         config = self.__merge_configs(config_content, secrets_config_content)
 
+        # rename 'type' to 'vtype' as 'type' is a builtin function
```

Comment:
This appears to have consequences on the JSON schema. Is this correct?

I don't think that implementation details (like what keywords exist in the programming language used by Connaisseur) should be exposed to the JSON schema. Have you considered working around this by using `**kwargs` instead of a static `type` argument, and mimicking the required-ness of the `type` (`vtype`) argument in below methods by inspecting `kwargs`? If you did consider it, which were the reasons to not pursue it?

Reply:
It does have consequences because I rename before the validation method below. Could rename afterwards and the schema itself would not be affected, but we'd still be working with `vtype` later. Might be better from a validate-before-parsing PoV and leav3 the schema intact. What do you think?
Have not considered using kwargs, but think that may feel somewhat obscure if the validator type comes out of the dark. Also not entirely sure if and how that would work. To be fair though, changing to `vtype` required supprisingly little change...

Reply:
I find it problematic that the schema (which requires `vtype`) does not reflect the input format because the input is modified before the schema is applied. This is going to be very confusing for anyone looking at the schema and configuration files, without looking at the code.

So, I'd definitely prefer validating the schema before making the modification.

My favorite solution would be to not modify the configuration. That's easily achieved by replacing `Validator.__new__` with
```python
    def __new__(cls, **kwargs):
        validator_type = kwargs.pop("type")
        try:
            return cls.class_map[validator_type](**kwargs)
        except KeyError:
            msg = f"{validator_type} is not a supported validator."
            raise NoSuchClassError(message=msg)  # pylint: disable=raise-missing-from
```

This way, you don't need several places to be aware of the matter.

Reply:
True. Like the idea!